internal/controller/postgrescluster/postgres.go

```diff
@@ -620,6 +621,12 @@ func (r *Reconciler) reconcilePostgresDataVolume(
 
 	pvc.Spec = instanceSpec.DataVolumeClaimSpec
 
+	r.determineVolumeSize(ctx, cluster, pvc, instance.Name, instanceSpec.Name)
+
+	// Clear any set limit before applying PVC. This is needed to allow the limit
+	// value to change later.
```

Comment:
:question: What is it about setting to nil here that allows the limit to change later?

Reply:
The `DataVolumeClaimSpec`, with the exception of the request value, is immutable once the PVC is bound. Effectively we're managing the limit ourselves but never applying it.